fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/QueryTransformer.java

```diff
@@ -292,10 +292,22 @@ private OptExprBuilder window(OptExprBuilder subOpt, List<AnalyticExpr> window)
         }
 
         final ExpressionMapping expressionMapping = subOpt.getExpressionMapping();
-        boolean allColumnRef = projectExpressions.stream()
-                .map(expression -> SqlToScalarOperatorTranslator.translate(expression, expressionMapping,
-                        columnRefFactory))
-                .allMatch(ScalarOperator::isColumnRef);
+        boolean allColumnRef = true;
+        Map<Expr, ColumnRefOperator> tempMapping = new HashMap<>();
+        for (Expr expression : projectExpressions) {
+            ScalarOperator operator = SqlToScalarOperatorTranslator.translate(expression, expressionMapping,
+                    columnRefFactory);
+            if (!operator.isColumnRef()) {
+                allColumnRef = false;
+                tempMapping.clear();
+                break;
+            } else {
+                tempMapping.put(expression, (ColumnRefOperator) operator);
+            }
+        }
+        if (allColumnRef) {
+            expressionMapping.getExpressionToColumns().putAll(tempMapping);
+        }
```

Comment:
Could you explain why this helps improve the performance?

Reply:
Before: iterate all `projectExpressions #translate ` and then check whether all are column refs
After: iterate one by one and check whether it's column ref, it will shortcut if not.


Can you encapsulate it into a function to be better? maybe use `Optional<Map<Expr, ColumnRefOperator>>` only if all are columns?
```
    for (Expr expression : projectExpressions) {
            ScalarOperator operator = SqlToScalarOperatorTranslator.translate(expression, expressionMapping,
                    columnRefFactory);
            if (!operator.isColumnRef()) {
                allColumnRef = false;
                tempMapping.clear();
                break;
            } else {
                tempMapping.put(expression, (ColumnRefOperator) operator);
            }
        }
```

Reply:
>          List<ScalarOperator> partitions = new ArrayList<>();
            for (Expr partitionExpression : windowOperator.getPartitionExprs()) {
                ScalarOperator operator = SqlToScalarOperatorTranslator
                        .translate(partitionExpression, subOpt.getExpressionMapping(), columnRefFactory);
                partitions.add(operator);
            }

            List<Ordering> orderings = new ArrayList<>();
            for (OrderByElement orderByElement : windowOperator.getOrderByElements()) {
                ColumnRefOperator col =
                        (ColumnRefOperator) SqlToScalarOperatorTranslator
                                .translate(orderByElement.getExpr(), subOpt.getExpressionMapping(), columnRefFactory);
                orderings.add(new Ordering(col, orderByElement.getIsAsc(),
                        OrderByElement.nullsFirst(orderByElement.getNullsFirstParam())));
            }

Shortcut characteristic is one of optimization, but the main purpose is to reduce SqlToScalarOperatorTranslator.translate method calss  in reorderWindowOperator method

Reply:
hi @packy92 
plz review code

Reply:
I get your point.